### PR TITLE
Make it work on general MIPS64 processors

### PR DESCRIPTION
**Pull request by FlyGoat**
This PR addressed some build failures for ARCH_MIPS64, also provide a MIPS64_GENERIC target so we should be able to use OpenBLAS on Cavium processors.

### CODE REVIEW

File: cpuid_mips64.c [hunk 1]
```diff
@@ -136,9 +138,11 @@ void get_subarchitecture(void){
     printf("P6600");
   }else if(detect()==CPU_I6500){
     printf("I6500");
-  }else{
+  }else if(detect()==CPU_SICORTEX){
     printf("SICORTEX");
-  }
+  }else{
+    printf("MIPS64_GENEIRC");
```
Comment (martin-frbg) [inline cpuid_mips64.c]: small typo here :)

**Comment by martin-frbg** [comment]
Thanks - I have an Octeon3 patch lying around that I had meant to add "soon", but it does not really have much if anything Cavium-specific in it. (It should be possible to build on MIPS64 with TARGET=GENERIC already, but I suspect that would currently produce a 32bit build, might need to re-check that)

File: cpuid_mips64.c (hunk 1 above)
Comment (FlyGoat) [reply cpuid_mips64.c]: Good catch :-)
Fixed

**Comment by martin-frbg** [comment]
getarch still returns UNKNOWN cpu rather than your MIPS64_GENERIC type with this patch on Octeon

**Comment by FlyGoat** [comment]
> getarch still returns UNKNOWN cpu rather than your MIPS64_GENERIC type with this patch on Octeon

Fixed. We can safely assume all Linux MIPS system is safe to run MIPS64_GENERIC.

**Comment by martin-frbg** [comment]
now that I have merged this, I am wondering if there isn't considerable overlap with the venerable SICORTEX target, maybe that is the effectively generic target I remembered ?

**Comment by FlyGoat** [comment]
> now that I have merged this, I am wondering if there isn't considerable overlap with the venerable SICORTEX target, maybe that is the effectively generic target I remembered ?

Yeah probably… Though I can’t spot fallback to SICORTEX in code.

**Comment by martin-frbg** [comment]
cpuid_mips64 does it when there is a model name in /proc/cpuinfo that does not match any of the Loongson types. seems on your machine there is no model name entry in cpuinfo ? the octeon I used for testing is running BSD, I'll need to dig out my own (early octeon3 running linux) to check

**Pull request merged**